src/cpp/r/RSexp.cpp

```diff
@@ -1517,9 +1517,8 @@ SEXP createUtf8(const std::string& data, Protect* pProtect)
    SEXP strSEXP;
    pProtect->add(strSEXP = Rf_allocVector(STRSXP, 1));
 
-   SEXP charSEXP;
-   pProtect->add(charSEXP = Rf_mkCharLenCE(data.c_str(), data.size(), CE_UTF8));
-
+   // protection not required as long as we immediately assign into protected STRSXP
+   SEXP charSEXP = Rf_mkCharLenCE(data.c_str(), data.size(), CE_UTF8);
```

Comment:
Do you have any resources to better understand what this is doing here?

Reply:
The available documentation here is unfortunately kind of sparse; see e.g.

https://cran.r-project.org/doc/manuals/r-release/R-ints.html#index-mkCharLenCE
https://github.com/hadley/r-internals/blob/master/strings.md